src/main/java/org/springframework/retry/policy/PredicateRetryPolicy.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2006-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.retry.policy;
+
+import java.util.function.Predicate;
+
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.RetryPolicy;
+import org.springframework.retry.context.RetryContextSupport;
+
+/**
+ * A policy, that is based on {@link Predicate<Throwable>}. Usually, binary classification
+ * is enough for retry purposes. If you need more flexible classification, use
+ * {@link ExceptionClassifierRetryPolicy}.
+ *
+ * @author Morulai Planinski
```

Comment:
Yeah... Now I see your point.
This really should be much straightforward than trying to adapt `ExceptionClassifierRetryPolicy` or `ExpressionRetryPolicy`.

Please, add `@since 2.0.7`